vendor/models/base.py

```diff
@@ -91,9 +90,10 @@ def get_best_currency(self, currency=DEFAULT_CURRENCY):
         If no currency is provided as an argument it will default to the products's msrp default currency.
         If currency is provided but is not available in the product it will default to the products's msrp default currency.
         """
-        if currency in self.meta['msrp']:
+        available_currencies = set(AVAILABLE_CURRENCIES.keys()).intersect(self.meta['msrp'].keys())
```

Comment:
does `value['msrp'].keys()` need to be type converted to a set() too for the intersect?

Since this is the second location of the same code, might be good idea to make it into a single function that can be shared for consistency.

Reply:
added the util function is_currency_available